pinot-common/src/main/java/org/apache/pinot/common/function/scalar/JsonFunctions.java

```diff
@@ -321,17 +321,64 @@ private static void setValuesToMap(String keyColumnName, String valueColumnName,
       Map<String, String> objMap = (Map) obj;
       result.put(objMap.get(keyColumnName), objMap.get(valueColumnName));
     } else {
-      ObjectMapper mapper = new ObjectMapper();
       JsonNode mapNode;
       try {
-        mapNode = mapper.readTree(obj.toString());
+        mapNode = OBJECT_MAPPER.readTree(obj.toString());
       } catch (JsonProcessingException e) {
         throw new RuntimeException(e);
       }
       result.put(mapNode.get(keyColumnName).asText(), mapNode.get(valueColumnName).asText());
     }
   }
 
+  /**
+   * Returns all key paths in a JSON object up to the specified depth.
+   * Each key path is dot-separated for nested keys.
+   * @param jsonObj JSON object or string
+   * @param maxDepth Maximum depth to recurse (inclusive)
+   * @return Array of key paths (dot notation)
+   */
+  @ScalarFunction(nullableParameters = true)
+  public static List jsonKeys(@Nullable Object jsonObj, int maxDepth) {
```

Comment:
Why do we allow null json object? When any input is null, output should be null, so we don't need to annotate it with `nullableParameters`